### ISSUE TITLE
`Literal`'s display for non-special unicode chars is not valid python

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#2862]
### Summary

This wasn't a real problem before, but now with inlay hint inserting you can end up with invalid python code. I think it would also be nice for consistency if it was valid python instead of special casing inlay hint inserting. It also shouldn't be too hard to just use `\uxxxx` and `\Uxxxxxxxx` (famous last words).
```
PS ~>echo @'
```
```py
from typing import reveal_type
reveal_type("\1")
```
```
'@ > issue.py
```
```
PS ~>uvx ty check issue.py
```
```
info[revealed-type]: Revealed type
 --> issue.py:2:13
  |
1 | from typing import reveal_type
2 | reveal_type("\1")
  |             ^^^^ `Literal["\u{1}"]`
  |

Found 1 diagnostic
```
Example playground where inserting the inlay hint gets you invalid code:
https://play.ty.dev/bf79e6a6-aadd-4a90-9d08-e3024ee825bc
```py
x[: tuple[Literal["\u{1}"], str]] = ("\1", input())
```
```pycon
>>> from typing import Literal
... x: tuple[Literal["\u{1}"], str] = ("\1", input())
...
  File "<python-input-42>", line 2
    x: tuple[Literal["\u{1}"], str] = ("\1", input())
                     ^^^^^^^
SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 0-1: truncated \uXXXX escape
```
This also applies to byte strings.

### Version

ty 0.0.17 (8cec85718 2026-02-13) playground 729610acd
